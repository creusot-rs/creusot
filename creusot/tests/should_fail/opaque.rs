extern crate creusot_contracts;

// INTENT: Should produce an error message because we reference `priv_symbol` which is
// less visible than we are opaque.

pub mod x {
    use creusot_contracts::*;

    #[predicate]
    fn priv_symbol() -> bool {
        true
    }

    #[open(crate)]
    #[predicate]
    pub fn bad() -> bool {
        priv_symbol()
    }
}
