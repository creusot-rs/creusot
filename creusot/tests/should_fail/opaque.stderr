error[creusot]: Cannot make `"x::priv_symbol"` transparent in `"x::bad"` as it would call a less-visible item.
  --> opaque.rs:17:9
   |
17 |         priv_symbol()
   |         ^^^^^^^^^^^^^

error: aborting due to previous error

